File: backend/models/TrafficInfo.js
const mongoose = require("mongoose");

const trafficInfoSchema = new mongoose.Schema({
  time: {
    type: Date,
    required: true,
  },
  location: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "location",
    required: true,
  },
  delay: {
    type: Number,
    required: true,
  },
  status: {
    type: String,
    enum: ["brez zastojev", "zastoj", "nesreča", "delo na cesti"],
    required: true,
  },
});

const TrafficInfo = mongoose.model("TrafficInfo", trafficInfoSchema);

module.exports = TrafficInfo;

File: src/main/kotlin/model/TrafficInfo.kt
package model

import com.google.gson.annotations.SerializedName
import java.util.*

data class TrafficInfo (
    @SerializedName("_id") val id: String? = null,
    val location: String,
    val delay: Int,
    val status: String,
    val time : Date
){
    override fun toString(): String {
        return "TrafficInfo(id='$id',\n" +
                "location='$location',\n" +
                "delay=$delay,\n" +
                "status='$status'\n" +
                "time='$time')"
     }
}